runtimes/bundles/org.jboss.tools.rsp.server.wildfly/src/main/java/org/jboss/tools/rsp/server/wildfly/servertype/publishing/StandardJBossPublishController.java

```diff
@@ -63,23 +63,29 @@ protected AbstractJBossServerDelegate getDelegate() {
 	private boolean hasSuffix(String path, String suffix) {
 		return path.endsWith(suffix);
 	}
-	private boolean isSupportedDeployable(String path, String outputName, boolean mustExist) {
+	
+	private IStatus validateSupportedDeployable(String path, String outputName, boolean mustExist) {
 		if( path == null )
-			return false;
+			return new Status(IStatus.ERROR, Activator.BUNDLE_ID, 
+					NLS.bind("Server {0} does not support deployment with null path.", server));
 		
 		File f = new File(path);
 		// When removing a module, there's no reason it must exist
 		if( mustExist && !f.exists())
-			return false;
-		
+			return new Status(IStatus.ERROR, Activator.BUNDLE_ID, 
+					NLS.bind("Server {0} does not support deployments that do not exist in the filesystem.", server));
+
 		if( mustExist && !supportsExplodedDeployment() && !f.isFile())
-			return false;
+			return new Status(IStatus.ERROR, Activator.BUNDLE_ID, 
+					NLS.bind("Server {0} does not support exploded deployment, and deployment path is a directory.", server));
 		
 		for( int i = 0; i < supportedSuffix.length; i++ ) {
 			if( hasSuffix(outputName, supportedSuffix[i]))
-				return true;
+				return Status.OK_STATUS;
 		}
-		return false;
+		return new Status(IStatus.ERROR, Activator.BUNDLE_ID, 
+				NLS.bind("Server {0} does not support deployment of resources without an approved suffix. " + 
+		"Use the " + ServerManagementAPIConstants.DEPLOYMENT_OPTION_OUTPUT_NAME + " deployment option to override the output name.", server));
```

Comment:
using ```server``` as parameter in string substitution will lead to a java object reference being used in the error message:
![image](https://user-images.githubusercontent.com/25126/58098947-dfd6e400-7bda-11e9-85ac-9be8642ea6e5.png)

Reply:
fixed it here: https://github.com/redhat-developer/rsp-server/pull/368